### ISSUE TITLE
why not use "blocking queries" with consul?

### ISSUE THREAD

**Issue by SnailDev** [ThreeMammals/Ocelot#374]
## Expected Behavior / New Feature

> Many endpoints in Consul support a feature known as "[blocking queries](https://www.consul.io/api/index.html#blocking-queries)". A blocking query is used to wait for a potential change using long polling. Not all endpoints support blocking, but each endpoint uniquely documents its support for blocking queries in the documentation.

blocking queries  could notify ocelot the lastest configuration immediately (when store configuration in consul) and  the lastest health services immediately(when service discovery and load balancer).

**Comment by SnailDev** [comment]
I have read source of ocelot, with test,  and I found if use service discovery and load balancer, ocelot will get health reroute server every request and it will affect performance. So I think memorycache and block queries would be useful.

**Comment by TomPallister** [comment]
I have merged a new feature that will allow users to specify if they want to poll consul

**Comment by TomPallister** [comment]
Released in 7.0.5

**Comment by SnailDev** [comment]
@TomPallister I have read your solution. but there are two problems:
1. PollingConsulServiceDiscoveryProvider init on the first request and _timer poll services from consul, so the first request will be fail.
2. _timer poll just run once. I logged in poll method and I found the log print just on first request.
```C#
private async Task Poll()
{
    _logger.LogInformation("Started polling consul");
    _services = await _consulServiceDiscoveryProvider.Get();
     _logger.LogInformation("Finished polling consul");
}
```

please confirm it, thank you very much.
